datalad/cmd.py

```diff
@@ -91,10 +90,19 @@ async def run_async_cmd(loop, cmd, protocol, stdin, protocol_kwargs=None,
         stdout=asyncio.subprocess.PIPE if protocol.proc_out else None,
         stderr=asyncio.subprocess.PIPE if protocol.proc_err else None,
     )
+    # encode command as utf-8 since on systems with locale not utf-8
+    # (e.g. CentOS based discovery HPC) encoding might fail although
+    # if encoded explicitly - works just fine
+    if on_windows:
+        # but windows does not allow for bytes here, so we are doomed to not
+        # do that: TypeError: bytes args is not allowed on Windows
+        encode_ = lambda s: s
+    else:
+        encode_ = lambda s: s.encode('utf-8')
     if isinstance(cmd, str):
-        proc = loop.subprocess_shell(factory, cmd, **kwargs)
+        proc = loop.subprocess_shell(factory, encode_(cmd), **kwargs)
     else:
-        proc = loop.subprocess_exec(factory, *cmd, **kwargs)
+        proc = loop.subprocess_exec(factory, *map(encode_, cmd), **kwargs)
```

Comment:
this change (from the 2nd commit) is the most questionable in this PR.  According to tests it seems to do the right thing, and the fact that it otherwise fails on "encode" step suggests that a similar action is done by Python std library itself.  But here we would "hardcode" us into "utf-8" as the "one and only" we support.
An alternative would be to address failing tests by making them skip if system locale is not utf-8 or not use unicode (since might fail) in the test values -- but the point of some of those tests is exactly that -- test that unicode could be provided, and according to results of the test run on those systems with non utf-8 locale -- it is possible (with this explicit `encode` here).

WDYT?

Reply:
no longer pertinent here - decided it is worth a separate PR/discussion: #5961